posthog/models/person/person.py

```diff
@@ -173,3 +177,40 @@ class Meta:
 
     oldest_event: models.DateTimeField = models.DateTimeField()
     version: models.BigIntegerField = models.BigIntegerField(null=True, blank=True)
+
+
+def get_distinct_ids_for_subquery(person: Person | None, team: Team) -> List[str]:
+    """_summary_
+    Fetching distinct_ids for a person from CH is slow, so we
+    fetch them from PG for certain queries. Therfore we need
+    to inline the ids in a `distinct_ids IN (...)` clause.
+
+    This can cause the query to explode for persons with many
+    ids. Thus we need to limit the amount of distinct_ids we
+    pass through.
+
+    The first distinct_ids should contain the real distinct_ids
+    for a person and later ones should be associated with current
+    events. Therefore we union from both sides.
+
+    Many ids are usually a sign of instrumentation issues
+    on the customer side.
+    """
+    first_ids_limit = 100
+    last_ids_limit = MAX_LIMIT_DISTINCT_IDS - first_ids_limit
+
+    if person is not None:
+        first_ids = (
+            PersonDistinctId.objects.filter(person=person, team=team)
```

Comment:
minor nit: Even if it were 1 million+ distinct ids, they'll easily fit into memory, so instead of 2 requests to the db, you can do the cutting in python, since for most users anyway the list will be small enough that the 2nd query does nothing really (same list of ids)

Reply:
I union together the querysets further below, so that's just a single request :) `first_ids.union(last_ids)`.

Reply:
missed that, nice :)